src/qml/Stickers/StickersPicker.qml

```diff
@@ -16,77 +16,213 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import QtQuick 2.3
+import QtQuick 2.9
 import Ubuntu.Components 1.3
+import Ubuntu.Components.Popups 1.3
 import messagingapp.private 0.1
 
+import ".." //ContentImport
+
 FocusScope {
     id: pickerRoot
+
+    property bool expanded: false
+    readonly property int packCount: stickerPacksModel.count
+
+    height: units.gu(30)
+
     signal stickerSelected(string path)
 
+
     Component.onCompleted: {
         StickersHistoryModel.databasePath = dataLocation + "/stickers/stickers.sqlite"
         StickersHistoryModel.limit = 10
     }
 
-    property bool expanded: false
-    readonly property int packCount: stickerPacksModel.count
+    onStickerSelected:  {
+        StickersHistoryModel.add(toSystemPath(path))
+    }
+
+    //backend need filepath without "file://" if any
+    function toSystemPath(path) {
+        return path.replace('file://', '')
+    }
+
+    function importStickerRequested(packName) {
+        contentImporter.requestPicture()
+        contentImporter.contentReceived.connect(function(contentUrl) {
+            stickerPacksModel.addSticker(packName, toSystemPath(String(contentUrl)))
+        })
+    }
 
-    // FIXME: try to get something similar to the keyboard height
-    // FIXME: animate the displaying
-    height: expanded ? units.gu(30) : 0
-    opacity: expanded ? 1 : 0
-    visible: opacity > 0
-
-    Connections {
-        target: Qt.inputMethod
-        onVisibleChanged: {
-            if (Qt.inputMethod.visible && oskEnabled) {
-                pickerRoot.expanded = false
+    StickersPackModel {
+        id: stickerPacksModel
+        stickerPath: dataLocation + "/stickers"
+
+        onCountChanged: console.log('sps count', count);
```

Comment:
Are the console.log's in this file still needed, or were they only for your testing?